src/converse-register.js

```diff
@@ -241,6 +245,25 @@
                     return false;
                 },
 
+                chooseProvider: function (ev) {
```

Comment:
Javascript is very forgiving when it comes to parameters passed to methods or functions. So `onProviderChosen` doesn't *have* to be called with an event as first parameter, it just currently assumes that it will. This is something that can be fixed with a bit of refactoring.

Alternatively what you can do is to have one method for the `submit` event, which then simply calls another method which then contains the common code from `chooseProvider` and `onProviderChosen`.